openapi/controllers/main.py

```diff
@@ -39,3 +49,77 @@ def copy(obj, **d):
         }})
 
         return result
+
+
+class OAS(http.Controller):
+
+    @http.route('/api/v1/<namespace_name>/swagger.json',
+                type='http', auth='public', csrf=False)
+    def OAS_json_spec_download(self, namespace_name, **kwargs):
+        namespace = http.request.env['openapi.namespace'].sudo().search([('name', '=', namespace_name)])
+        if not namespace:
+            raise werkzeug.exceptions.NotFound()
+
+        current_host = http.request.env['ir.config_parameter'].sudo().get_param('web.base.url')
+        parsed_current_host = urlparse.urlparse(current_host)
+
+        spec = {
```

Comment:
I think we have to move this part to a spec generating method in the 'openapi.namespace' model?